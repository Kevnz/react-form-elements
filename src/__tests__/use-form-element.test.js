import React, { createRef } from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import useFormElement from '../utils/use-form-element'

describe('useFormElement hook', () => {
  afterEach(() => {
    cleanup()
  })

  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const val = 'My Input Value'
      const testId = 'test-hook-1'
      const TestInput = () => {
        const ref = createRef()
        const { value, handleChange, inputRef } = useFormElement(val, ref)

        return (
          <input
            data-testid={testId}
            name="hookedInput"
            type="text"
            value={value}
            onChange={handleChange}
            ref={inputRef}
          />
        )
      }

      const { getByTestId, container } = render(<TestInput />)

      expect(container.firstChild).toMatchSnapshot()
      const input = getByTestId(testId)
      expect(input.value).toBe(val)
    })
    it('should update a value when changed', () => {
      const val = 'My Input Value'
      const testId = 'test-hook-2'
      const TestInput = () => {
        const ref = createRef()
        const { id, value, handleChange, inputRef } = useFormElement(val, ref)

        return (
          <input
            data-testid={testId}
            name="hookedInput"
            type="text"
            value={value}
            onChange={handleChange}
            ref={inputRef}
          />
        )
      }
      const { getByTestId, container } = render(<TestInput />)
      const input = getByTestId(testId)

      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.value).toBe('Updated Value')
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
