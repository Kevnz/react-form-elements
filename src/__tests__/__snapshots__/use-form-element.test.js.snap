// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFormElement hook handling input should set a value when passed in 1`] = `
<input
  data-testid="test-hook-1"
  name="hookedInput"
  type="text"
  value="My Input Value"
/>
`;

exports[`useFormElement hook handling input should update a value when changed 1`] = `
<input
  data-testid="test-hook-2"
  name="hookedInput"
  type="text"
  value="Updated Value"
/>
`;
